src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.DecisionDagRewriter.cs

```diff
@@ -217,8 +243,12 @@ public override BoundNode VisitObjectInitializerMember(BoundObjectInitializerMem
 
                 public override BoundNode VisitIndexerAccess(BoundIndexerAccess node)
                 {
-                    // Although property arguments with ref indexers are not declarable in C#, they may be usable
-                    if (!node.ArgumentRefKindsOpt.IsDefault)
+                    bool mightMutate =
+                        !node.ArgumentRefKindsOpt.IsDefault ||
+                        // We only need to check the get accessor because an assigment would cause _mightAssignSomething to be set to true in the caller
+                        MethodMayMutateReceiver(node.ReceiverOpt, node.Indexer.GetMethod);
```

Comment:
>node.ReceiverOpt [](start = 48, length = 16)

It doesn't look like ```MethodMayMutateReceiver``` is going to handle null as the first parameter.

Reply:
That can only occur for static methods, and `MethodMayMutateReceiver` guards for that case.

Reply:
> That can only occur for static methods, and MethodMayMutateReceiver guards for that case.

Not really. This can happen for some error scenarios. I believe we recently fixed a bug like that in a NullableWalker. I suggest we make the code robust, not make any assumptions and add a null check which would cost us nothing.

---
In reply to: [405757790](https://github.com/dotnet/roslyn/pull/42972#discussion_r405757790) [](ancestors = 405757790)

Reply:
You previously pushed back against strengthening lowering to support the lowering of erroneous code. Lowering is not called for code that violates the language rules. Until those change, I'm keeping this as is.

---
In reply to: [405764292](https://github.com/dotnet/roslyn/pull/42972#discussion_r405764292) [](ancestors = 405764292,405757790)